rpcserver.go

```diff
@@ -6157,6 +6162,38 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 
 			blindingRestrictions.NodeOmissionSet.Add(vertex)
 		}
+
+		blindingRestrictions.IncomingChainedChannels = append(
+			blindingRestrictions.IncomingChainedChannels,
+			blindCfg.IncomingChannelList...,
+		)
+
+		numChainedChannels :=
+			uint8(len(blindingRestrictions.IncomingChainedChannels))
+
+		// When selecting the blinded incoming channel list parameter
+		// the maximum number of hops is implictitly set.
+		if numChainedChannels > blindingRestrictions.NumHops {
+			rpcsLog.Warnf("Changing the num_blinded_hops "+
+				"from (%d) to (%d)",
+				blindingRestrictions.NumHops,
+				numChainedChannels)
+
+			blindingRestrictions.NumHops =
+				numChainedChannels
+		}
+
+		// The MinDistanceFromIntroNode must be greater than or equal to
+		// the number of hops specified on the chained channels.
+		minNumHops := blindingRestrictions.MinDistanceFromIntroNode
+		if minNumHops < numChainedChannels {
```

Comment:
Add a comment here like this:

`We need to make sure the MinimumPath is at least the size of the chained channel list because otherwise our pathfinder would return routes even smaller than our provided list (if available) but we want at least the path which we provided not a smaller one.`